app/controllers/users/totp_setup_controller.rb

```diff
@@ -7,10 +7,12 @@ def start
     end
 
     def new
-      if user_session[:new_totp_secret].nil?
+      if current_user.totp_enabled?
+        redirect_to profile_path, notice: t('errors.authenticator_app_already_enabled')
+      else
         user_session[:new_totp_secret] = current_user.generate_totp_secret
```

Comment:
I think we still want to not  generate new secrets every time the user loads this page.  Do we should check `new_totp_secret.nil?` here. And I guess we should add a test that reloading the page keeps the code the same.   If you want you could split this whole totp change into a separate PR?